keps/sig-architecture/20191023-conformance-without-beta.md

```diff
@@ -0,0 +1,167 @@
+production distributions are expected to increase in stability, reliability, and consistency.
+The Kubernetes conformance project is an important aspect of ensuring consistency across distributions.
+A production distribution should not be required to enable non-GA features or REST APIs in order to pass conformance tests.
+
+Beta dependencies can be grouped into several categories:
+* Kubernetes components requiring beta REST APIs with no stable alternatives (for example, kubelets use the beta CertificateSigningRequest endpoints for cert rotation)
+* Kubernetes components requiring beta REST APIs with stable alternatives (for example, kube-scheduler accidentally switched to using the beta Events REST API in 1.16)
+* Kubernetes components requiring behavior enabled by beta feature gates
+* Kubernetes conformance tests requiring calls to beta REST APIs
+* Kubernetes conformance tests exercising behavior enabled by beta feature gates
+
+For each category, we should identify and resolve existing dependencies, and prevent future occurrences.
+
+### Goals
+
+* Identify existing beta REST APIs and features required by Kubernetes components so they can be graduated or the dependencies removed
+* Identify existing beta REST APIs and features required by conformance tests so the tests can be rewritten to remove those dependencies
+* Prevent new beta REST APIs and features from being required by Kubernetes components or conformance tests
+* Demonstrate passing conformance tests on a cluster with all beta REST APIs and features disabled
+
+### Non-Goals
+
+* Resolve questions about conformance profiles
+* Resolve questions about GA-but-optional features
+* Forbid beta REST APIs or features from being enabled in conformant clusters by distributors
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a Kubernetes component developer, I cannot accidentally make changes
+that would require a production cluster to enable a beta API or feature
+in order to pass conformance tests.
+
+Attempts to make such a change would be caught/blocked by a CI job.
+
+#### Story 2
+
+As a Kubernetes conformance test author, I cannot accidentally write 
+(or promote to conformance) tests that require use of beta REST APIs.
+
+Attempts to make such a change would be caught/blocked by a CI job.
+
+#### Story 3
+
+As a Kubernetes distribution maintainer, I can pass conformance tests
+without enabling beta REST APIs or features.
+
+A canonical CI job running against Kubernetes master and release branches
+would demonstrate that it is possible to pass conformance tests with no
+beta REST APIs or features enabled.
+
+## Design Details
+
+1. Make it possible to easily disable all beta REST APIs and features
+  * Add support to the Kubernetes API server with `--runtime-config=api/beta=false`
+    * Parallels use of `--runtime-config=api/all=false`
+  * Add support to all components with `--feature-gates=AllBeta=false`
```

Comment:
Same here - IMO `AllBeta=false` should imply `AllAlpha=false`, as alpha features should be able to depend on beta features (but not vice-versa)

Reply:
I'd rather keep the levers independent